src/scripts/main.js

```diff
@@ -1,3 +1,120 @@
 'use strict';
 
-// write code here
+'use strict';
+
+const table = document.querySelector('table');
+let sortOrder = 'asc';
+
+table.tHead.addEventListener('click', (evt) => {
+  const target = evt.target.closest('th');
+
+  if (target) {
+    const index = Array.from(target.parentNode.children).indexOf(target);
+    const rows = Array.from(table.tBodies[0].rows);
+
+    const sortedRows = rows.sort((rowA, rowB) => {
+      const cellA = rowA.cells[index].textContent.trim();
+      const cellB = rowB.cells[index].textContent.trim();
+
+      if (!isNaN(cellA) && !isNaN(cellB)) {
+        return sortOrder === 'asc' ? cellA - cellB : cellB - cellA;
+      } else {
+        return sortOrder === 'asc'
+          ? cellA.localeCompare(cellB)
+          : cellB.localeCompare(cellA);
+      }
+    });
+
+    sortOrder = sortOrder === 'asc' ? 'desc' : 'asc';
+
+    sortedRows.forEach((row) => table.tBodies[0].appendChild(row));
+  }
+});
+
+table.tBodies[0].addEventListener('click', (evt) => {
+  const targetRow = evt.target.closest('tr');
+
+  if (targetRow) {
+    const rows = table.tBodies[0].querySelectorAll('tr');
+
+    rows.forEach((row) => row.classList.remove('active'));
+    targetRow.classList.add('active');
+  }
+});
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (evt) => {
+  evt.preventDefault();
+
+  const employeeName = form.elements['name'].value.trim();
+  const position = form.elements['position'].value.trim();
+  const office = form.elements['office'].value.trim();
+  const age = +form.elements['age'].value.trim();
+  const salary = +form.elements['salary'].value.trim();
```

Comment:
Consider adding validation to ensure the salary is a positive number. Currently, negative or zero values are not checked, which might lead to incorrect data being added to the table.